aiohttp/web_utils.py

```diff
@@ -0,0 +1,57 @@
+import asyncio
+
+from aiohttp.web_exceptions import HTTPMovedPermanently, HTTPNotFound
+from aiohttp.web_urldispatcher import SystemRoute
+
+__all__ = ('normalize_path_middleware',)
+
+
+def normalize_path_middleware(*, merge_slashes=True, append_slash=True,
+                              redirect_class=HTTPMovedPermanently):
+
+    @asyncio.coroutine
+    def factory(app, handler):
+
+        @asyncio.coroutine
+        def middleware(request):
+            try:
+                return (yield from handler(request))
+            except HTTPNotFound as exc:
```

Comment:
Hmm... catching `HTTPNotFound` here may lead to unexpected behavior, imagine the following use case:
```python
app.router.add_route("GET", "/root/resource", handler1)
app.router.add_route("GET", "/root/resource/{tail:.*}", handler2)

def handler1(req):
    if not has_permission(req):
        raise web.HTTPNotFound()   # we don't want to expose this resource
    # ....

def handler2(req):
    pass
```
In this case application will expose `/root/resource/` url (that it is exsist) and also it may be hard to debug such cases.